src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java

```diff
@@ -171,13 +171,18 @@ public static UpdateRequestQueue using(Clients clients,
   public CompletableFuture<Result<LoanAndRelatedRecords>> onCheckOut(
     LoanAndRelatedRecords relatedRecords) {
 
-    return onCheckOut(relatedRecords.getRequestQueue())
+    return onCheckOut(relatedRecords.getRequestQueue(), relatedRecords.getItem().getItemId())
       .thenApply(result -> result.map(relatedRecords::withRequestQueue));
   }
 
-  private CompletableFuture<Result<RequestQueue>> onCheckOut(RequestQueue requestQueue) {
+  private CompletableFuture<Result<RequestQueue>> onCheckOut(RequestQueue requestQueue,
+    String itemId) {
+
     if (requestQueue.hasOutstandingFulfillableRequests()) {
       Request firstRequest = requestQueue.getHighestPriorityFulfillableRequest();
+      if (!firstRequest.getItemId().equals(itemId)) {
```

Comment:
firstRequest.getItemId() - can this be null?

Reply:
updated